### PR TITLE
Updates related to fix the attributes issue while creating the fortify application

### PR DESCRIPTION
**Pull request by msenmurugan**
Updates related to fix the attributes issue while creating the fortify application
Renamed the exception method name

### CODE REVIEW

**Comment by robpacheco** [comment]
Why is almost everything static?

File: src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPushThread.java [hunk 1]
```diff
@@ -80,8 +78,11 @@
 
     private final static Logger logger = Logger.getLogger(BlackDuckFortifyPushThread.class);
 
+    private SpringConfiguration springConfiguration;
```
Comment (robpacheco) [inline src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPushThread.java]: Can this be final?

Comment (robpacheco) [inline src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPushThread.java]: Actually, is this class - BlackDuckForfifyPushThread a spring bean as well? I wouldn't expect a class to make an instance of the spring config. I'd expect the spring config to send its beans to the constructor of other beans. If this class isn't a bean, some bean must make it, in which case, i'd send the required instances to that bean and into this constructor.

File: src/main/java/com/blackducksoftware/integration/fortify/batch/util/HubServices.java [hunk 2]
```diff
@@ -53,7 +53,11 @@
 
     private final static Logger logger = Logger.getLogger(HubServices.class);
 
-    private static HubServicesFactory hubServicesFactory = RestConnectionHelper.createHubServicesFactory();
+    private HubServicesFactory hubServicesFactory;
```
Comment (robpacheco) [inline src/main/java/com/blackducksoftware/integration/fortify/batch/util/HubServices.java]: Can this be final?

File: src/main/java/com/blackducksoftware/integration/fortify/batch/util/MappingParser.java [hunk 3]
```diff
@@ -62,8 +70,16 @@
 
     private final static String Q_connector = "+and+";
 
+    private final static String DYNAMIC_SCAN_REQUEST = "DYNAMIC_SCAN_REQUEST";
+
     private final static Logger logger = Logger.getLogger(MappingParser.class);
 
+    private SpringConfiguration springConfiguration;
```
Comment (robpacheco) [inline src/main/java/com/blackducksoftware/integration/fortify/batch/util/MappingParser.java]: Similar comment here as to the other class that used SpringConfig directly.

File: src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPushThread.java (hunk 1 above)
Comment (msenmurugan) [inline src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPushThread.java]: * Added final for HubServiceFactory and Spring Configuration
* Passed the Spring configuration as a constructor to all required
beans.

File: src/main/java/com/blackducksoftware/integration/fortify/batch/util/HubServices.java (hunk 2 above)
Comment (msenmurugan) [inline src/main/java/com/blackducksoftware/integration/fortify/batch/util/HubServices.java]: * Added final for HubServiceFactory

File: src/main/java/com/blackducksoftware/integration/fortify/batch/util/MappingParser.java (hunk 3 above)
Comment (msenmurugan) [inline src/main/java/com/blackducksoftware/integration/fortify/batch/util/MappingParser.java]: * Added final for Spring Configuration

File: src/main/java/com/blackducksoftware/integration/fortify/batch/util/MappingParser.java [hunk 4]
```diff
@@ -62,8 +70,16 @@
 
     private final static String Q_connector = "+and+";
 
+    private final static String DYNAMIC_SCAN_REQUEST = "DYNAMIC_SCAN_REQUEST";
+
     private final static Logger logger = Logger.getLogger(MappingParser.class);
 
+    private final SpringConfiguration springConfiguration;
```
Comment (robpacheco) [inline src/main/java/com/blackducksoftware/integration/fortify/batch/util/MappingParser.java]: Closer.. but nothing should take a SpringConfiguration, it should take the beans made from the SpringConfiguration. So for instance, this constructor should take: 

- an instance of FortiyApplicationVersionApi

Comment (msenmurugan) [inline src/main/java/com/blackducksoftware/integration/fortify/batch/util/MappingParser.java]: I have removed the Spring Configuration file and moved all the bean definition to the BlackDuckFortifyJobConfig where I have passed the required dependencies to the appropriate constructors. Can you please review it.

**Pull request merged**